### PR TITLE
Frontend Assessment 3 complete

### PR DESCRIPTION
**Pull request by asish918**
## Drift Video
[Drift Video](https://video.drift.com/v/abQgvXGDAy5qusAI2cQhfIySMWvRd5pnwKYfnfGBJwjo)

## Screenshots

Desktop View
screencapture-localhost-5173-2024-03-01-13_24_13

screencapture-localhost-5173-2024-03-01-13_25_33

Screenshot 2024-03-01 at 13 27 01

Mobile View
screencapture-localhost-5173-2024-03-01-13_36_54

screencapture-localhost-5173-2024-03-01-13_37_36

### CODE REVIEW

**Comment by asish918** [comment]
| Metric | Value |
| ------ | ----- |
| alert_status | <b>:white_check_mark: OK</b> |
| bugs | 0 |
| code_smells | 7 |
| reliability_rating | A (0 Bugs)<br/>Note: A being best, E being worst |
| security_rating | A (0 Vulnerabilities)<br/>Note: A being best, E being worst |
| vulnerabilities | 0 |

**Comment by Nitesh2503** [comment]
- Go through the code smells
- The commit message's format must be proper
- Use separate files for the interfaces, constants, etc.

File: frontend/assessments/3/src/components/BoxSelection.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    @media (max-width: 504px) {
+        text-align: center;
+    }
+`
+
+interface BoxSelectionProps {
+    type: Field
+}
```
Comment (Nitesh2503) [inline frontend/assessments/3/src/components/BoxSelection.tsx:34]: Could be stored in the different fike

File: frontend/assessments/3/src/components/BoxSelection.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                        </Selection>
+                    ))}
+                </SelectContainer>
+            }
+
+            {
+                type === "add-ons" && userData.activeRoom ?
+                    <SelectContainer>
+                        {roomData[userData.activeRoom?.id - 1].addOns.map(addOn => (
+                            <Selection onClick={() => selectAddOn(addOn)} $selected={userData.addOns?.includes(addOn)}>
```
Comment (Nitesh2503) [inline frontend/assessments/3/src/components/BoxSelection.tsx:69]: Format the code properly

File: frontend/assessments/3/src/components/SelectField.tsx [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+    height: 70px;
+    display: flex;
+    align-items: center;
+    padding: 20px;
+    margin-bottom: 20px;
+`
+
+const Price = styled.div`
+    font-size: 1.5rem;
+`
```
Comment (Nitesh2503) [inline frontend/assessments/3/src/components/SelectField.tsx:31]: Store in the separate file

**Pull request closed without merging**